### PR TITLE
tests: register all kind of tests for sqrt, log and abs

### PR DESCRIPTION
**Pull request by parth121101**

### CODE REVIEW

**Comment by parth121101** [comment]
@certik , can you check this and let me know if this is what exactly required for testing all the intrinsic functions?

File: integration_tests/intrinsics_01.f90 [hunk 1]
```diff
@@ -1,26 +1,74 @@
...
+    c(1) = -1._dp
+    c(2) = -1._dp
+    c(3) = -1._dp
+    c(4) = -1._dp
+    call random_number(c)
+    do i = 1, 4
+        if (c(i) < 0._dp) error stop
+        if (c(i) > 1._dp) error stop
+    end do
 end
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_01.f90:74]: I think we should also check everything for single precision too just like you did it for dp.

File: integration_tests/intrinsics_01.f90 [hunk 2]
```diff
@@ -1,26 +1,74 @@
...
+    real(dp) :: log_run(3) = [0.875_dp, 3.367_dp, 2.140_dp]
+    real(dp) :: abs_run(3) = [-723.182_dp, 9.124_dp, -123.145_dp]
+    real(dp) :: sqrt_res(3), log_res(3), abs_res(3)
+    real(dp) :: expected1(3) = [2.38285542994114508e+00_dp, 2.98709223158576087e+01_dp, 8.50364627674505336e+00_dp]
+    real(dp) :: expected2(3) = [-0.13353139262452263_dp, 1.2140221401789375_dp, 0.76080582903376015_dp]
+    real(dp) :: expected3(3) = [7.23182000000000000e+02_dp, 9.12400000000000000e+00_dp, 1.23145000000000000e+02_dp]
+
+    real(dp), parameter :: res1(3) = sqrt([5.678_dp, 892.272_dp, 72.312_dp])
+    real(dp), parameter :: res2(3) = log([0.875_dp, 3.367_dp, 2.140_dp])
+    real(dp), parameter :: res3(3) = abs([-723.182_dp, 9.124_dp, -123.145_dp])
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_01.f90:15]: This compile time test can be done for scalars too.

>For each new elemental intrinsic, we need to add 4 tests: compile time/runtime scalar/array (4 combinations).

File: integration_tests/intrinsics_01.f90 (hunk 1 above)
Comment (parth121101) [reply integration_tests/intrinsics_01.f90:74]: Sure @HarshitaKalani I will add those,

**Comment by HarshitaKalani** [comment]
>@HarshitaKalani Can you please look into this PR, the tests passes in my system, but fails in linux. It would be really helpful if you could check this!

I don't think that is the issue.
Just run the following command and you'll realise the issue:
```
$ ./integration_tests/run_tests.py -b fortran
```
The issue is with asr to fortran and I guess it is because of: #4052

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
This adds array compile time tests, which is good.

As a general guideline, since you had to remove "fortran", it's better to just add this as a new test and not put "fortran" in, that way we are keeping the old "fortran" test.

**Comment by parth121101** [comment]
> This adds array compile time tests, which is good.
> 
> As a general guideline, since you had to remove "fortran", it's better to just add this as a new test and not put "fortran" in, that way we are keeping the old "fortran" test.

@certik I can close this PR and can open a new PR with this changes. Will that work fine ?

**Comment by certik** [comment]
Sure, go ahead and create a new PR or update this PR to cleanly add the new test without removing old tests.

**Comment by parth121101** [comment]
Continuing this PR at #4142

**Pull request closed without merging**